socialdistribution/api/views.py

```diff
@@ -0,0 +1,113 @@
+from django.http import HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+
+from profiles.models import Author
+from posts.models import Post, Comment
+
+from .utils import post_to_dict, comment_to_dict
+
+
+# Create your views here.
+@csrf_exempt
+def all_posts(request):
+    # this view only accepts GET, 405 Method Not Allowed for other methods
+    if request.method != "GET":
+        return HttpResponse(f"Method {request.method} not allowed", status=405)
+
+    # get a list of all "PUBLIC" visibility posts on our node
+    public_posts = Post.objects.filter(visibility='PUBLIC')
+
+    # response body - to be converted into JSON and returned in the response
+    response_body = {
+        "query" : "posts",
+        "count" : public_posts.count(),
+        "size"  : "IMPLEMENT PAGINATION",
+        "next"  : "IMPLEMENT PAGINATION",
+        "previous" : "IMPLEMENT PAGINATION",
+        "posts" : [post_to_dict(post) for post in public_posts]
+    }
+
+    return JsonResponse(response_body)
+
+@csrf_exempt
+def single_post(request, post_id):
+    posts = Post.objects.filter(id=post_id)
+    
+    # GET a post which doesn't exist - 404 Not Found
+    if request.method == "GET" and posts.count() == 0:
+        return HttpResponse("That post does not exist", status=404)
+
+    # POST (insert) a post which already exists - 403 Forbidden
+    if request.method == "POST" and posts.count() > 0:
+        return HttpResponse("That post already exists", status=403)
```

Comment:
I would use "You do not have the permission into this post" or something similar to this.
Since 403 is Forbidden

Reply:
Even if the user has permission to edit the post, the requirements state that `POST` can only insert new posts, not update existing ones. So telling them they don't have permission doesn't really make sense.